Node 16 can not  install  zmq 5.2.0

**Describe the bug**
I can install zmq@5.2.8 with node.js v16,but can't install zmq@5.2.0 with node.js v16. the error message is below. And furthur, i can install zmq@5.2.0 and zmq@5.2.8 with node.js V10.
**Reproducing**
```
ERR! code 1
npm ERR! path C:\Users\birui\Desktop\monitor\node_modules\zeromq
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node scripts/prebuild-install.js || (node scripts/preinstall.js && node-gyp rebuild)
npm ERR! Downloading libzmq for Windows
npm ERR! Download finished
npm ERR! �ڴ˽��������һ������һ����Ŀ����Ҫ���ò������ɣ������ӡ�/m�����ء�
npm ERR! C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.targets(67,5): error MSB8020: �޷��ҵ� v140 �����ɹ���(ƽ̨���߼� =��v140��)����Ҫʹ�� v140 ���ɹ��߽������ɣ��밲װ v140 ���ɹ��ߡ����ߣ�������������ǰ Visual Studio ���ߣ���ʽ��ͨ��ѡ��
��Ŀ���˵����Ҽ������ý��������Ȼ��ѡ���ض��������Ŀ�ꡱ�� [C:\Users\birui\Desktop\monitor\node_modules\zeromq\build\zmq.vcxproj]
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@7.1.2
npm ERR! gyp info using node@16.4.2 | win32 | x64
npm ERR! gyp info find Python using Python version 3.8.1 found at "C:\Users\birui\AppData\Local\Programs\Python\Python38\python.exe"
npm ERR! gyp info find VS using VS2017 (15.9.28307.1585) found at:
npm ERR! gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community"
npm ERR! gyp info find VS run with --verbose for detailed information
npm ERR! (node:13208) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
npm ERR! (Use `node --trace-deprecation ...` to show where the warning was created)
npm ERR! gyp info spawn C:\Users\birui\AppData\Local\Programs\Python\Python38\python.exe
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   'C:\\Users\\birui\\AppData\\Local\\nvs\\node\\16.4.2\\x64\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'msvs',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\birui\\Desktop\\monitor\\node_modules\\zeromq\\build\\config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\birui\\AppData\\Local\\nvs\\node\\16.4.2\\x64\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\birui\\AppData\\Local\\node-gyp\\Cache\\16.4.2\\include\\node\\common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=C:\\Users\\birui\\AppData\\Local\\node-gyp\\Cache\\16.4.2',
```

**Expected behavior**
A clear and concise description of what you expected to happen.

**Tested on**
 - OS: [e.g.  Windows 10
 - ZeroMQ.js version: 5.2.0
 - msvs_version:2017

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -19,12 +19,12 @@ jobs:
           - macos-latest
           - windows-latest
         node_version:
-          - 14
+          - 18
         node_arch:
           - x64
         include:
           - os: windows-latest
-            node_version: 14
+            node_version: 18
             node_arch: x86
           # - os: macos-11.0
           #   node_version: 15
@@ -53,10 +53,3 @@ jobs:
         uses: actions/upload-artifact@v2
         with:
           path: ./prebuilds
-
-  Skip:
-    if: contains(github.event.head_commit.message, '[skip ci]')
-    runs-on: ubuntu-latest
-    steps:
-      - name: Skip CI 🚫
-        run: echo skip ci

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -15,25 +15,25 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-20.04
-          - macos-latest
-          - windows-latest
+          - ubuntu-22.04
+          - macos-11
+          - windows-2022
         node_version:
           - 18
         node_arch:
           - x64
         include:
-          - os: windows-latest
+          - os: windows-2022
             node_version: 18
             node_arch: x86
-          # - os: macos-11.0
-          #   node_version: 15
+          # - os: macos-11
+          #   node_version: 18
           #   node_arch: arm64
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
 
       - name: Install Node
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node_version }}
           architecture: ${{ matrix.node_arch }}

--- a/binding.gyp
+++ b/binding.gyp
@@ -1,6 +1,7 @@
 {
   'variables': {
     'zmq_external%': 'false',
+    'openssl_fips': '',
   },
   'targets': [
     {

--- a/package.json
+++ b/package.json
@@ -9,17 +9,17 @@
     "url": "https://github.com/zeromq/zeromq.js.git"
   },
   "dependencies": {
-    "nan": "2.14.2",
-    "node-gyp-build": "^4.2.3"
+    "nan": "2.17.0",
+    "node-gyp-build": "^4.5.0"
   },
   "devDependencies": {
     "electron-mocha": "^6.0.0",
-    "jsdoc": "^3.5.4",
     "mocha": "^5.0.0",
-    "nyc": "^12.0.2",
-    "prebuildify": "^4.1.2",
-    "semver": "^5.4.1",
-    "should": "^13.0.0"
+    "jsdoc": "^4.0.0",
+    "nyc": "^15.1.0",
+    "prebuildify": "^5.0.1",
+    "semver": "^7.3.8",
+    "should": "^13.2.3"
   },
   "engines": {
     "node": ">=6.0"

--- a/package.json
+++ b/package.json
@@ -27,8 +27,8 @@
   "scripts": {
     "install": "node-gyp-build || npm run build:libzmq",
     "build:libzmq": "node-gyp rebuild",
-    "prebuildify": "prebuildify -t 15.7.0 -t 14.16.1 -t 12.4.0 -t electron@5.0.13 -t electron@6.1.12 -t electron@7.3.3 -t electron@8.5.5 -t electron@9.4.4 -t electron@10.4.3 -t electron@11.4.3 --strip",
-    "prebuildify-ia32": "prebuildify --arch=ia32 -t 15.7.0 -t 14.16.1 -t 12.4.0 -t electron@5.0.13 -t electron@6.1.12 -t electron@7.3.3 -t electron@8.5.5 -t electron@9.4.4 -t electron@10.4.3 -t electron@11.4.3  --strip",
+    "prebuildify": "prebuildify -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 -t electron@21.0.1 --strip",
+    "prebuildify-ia32": "prebuildify --arch=ia32 -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 -t electron@21.0.1  --strip",
     "build:docs": "jsdoc -R README.md -d docs lib/*.js",
     "test": "mocha --expose-gc --slow 300",
     "test:electron": "electron-mocha --slow 300",

--- a/package.json
+++ b/package.json
@@ -27,8 +27,8 @@
   "scripts": {
     "install": "node-gyp-build || npm run build:libzmq",
     "build:libzmq": "node-gyp rebuild",
-    "prebuildify": "prebuildify -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 -t electron@21.0.1 --strip",
-    "prebuildify-ia32": "prebuildify --arch=ia32 -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 -t electron@21.0.1  --strip",
+    "prebuildify": "prebuildify -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 --strip",
+    "prebuildify-ia32": "prebuildify --arch=ia32 -t 18.12.1 -t 16.18.1 -t 14.21.1 -t 12.22.12 -t electron@11.5.0 --strip",
     "build:docs": "jsdoc -R README.md -d docs lib/*.js",
     "test": "mocha --expose-gc --slow 300",
     "test:electron": "electron-mocha --slow 300",

